### ISSUE TITLE
Enable OpenCL on Linux

### ISSUE THREAD

**Issue by goodyttoor** [laurent-clouet/sheepit-client#165]
I am rendering with OpenCL on Linux without any problem. Sheepit did not crash on GPU detection. May the problem only be Mac specific?

We probably can enable it on Linux and test it out. Thank you.

CPU: AMD Ryzen 2400G (Vega 11 GPU)
OS: Manjaro 17.1.12 64bit

**Comment by MCOfficer** [comment]
for future reference:
#44 [(comment)](https://github.com/laurent-clouet/sheepit-client/issues/44#issuecomment-416040206)

**Comment by Stubenhocker1399** [comment]
After trying various things to get my card to be detected I found that the card wasn't even attempted to be detected as per this line: https://github.com/laurent-clouet/sheepit-client/blob/760a5d871185ea53f35638dbc8f92444999814a1/src/com/sheepit/client/hardware/gpu/GPU.java#L42

It would be nice to at least warn the user that OpenCL is only being checked on windows and maybe even add a manual override for this check.

I'm wondering, would a try-catch not suffice here?

**Comment by re-l124c41plus** [comment]
Yeah, I'd like to use OpenCL on Linux because it would be a shame to leave all this GPU processing power unused. I think the sheepit cluster is overall losing a decent amount of potential GPU power here.

I can render using OpenCL in Blender just fine, so where is the issue? Is it in this project or some upstream library?
What can we do to help debug this and get this feature?

**Comment by Seltonu** [comment]
Bump! Would absolutely love this, I specifically went searching for ways to donate my CPU/GPU cycles and really wanted to find a distributed render farm to help out struggling artists. Please please let me put my Vega 56 to good use! How can we help debug?

**Comment by tuxun** [comment]
Bump. I compiled sheepit, removing the windows test, and it seems to work.  I use opencl with Amd driver on my RX580

**Comment by tuxun** [comment]
I've found [on the ArchLinux wiki](https://wiki.archlinux.org/index.php/AMDGPU) an usefull tip to disable P-state on amd gpu
```
echo "manual"> /sys/class/drm/card0/device/power_dpm_force_performance_level 
```
(its was "performance" by default)

afaik the last sheepit client handle opencl under linux, without modification, and my card seems way more stable in manual mode.

**Comment by tuxun** [comment]
Update:
On the date of 9th July 2020, amd released a new driver stack, compatible with ubuntu 20.20. 
I updated my O.S and my driver to the last version.

xserver only buged one time since 17 July.

I consider the stack stable enough for me, but how could we proceed for testing linux opencl on more AMD card?
https://www.amd.com/en/support/graphics/radeon-500-series/radeon-rx-500-series/radeon-rx-580

Is there someone else here wanting to help us to check the status of openCL?

SHoud we try to add a checkbox in the device selection panel setting  (maybe only if detected OS is linux), for enable opencl rendering ? a warning should be shown, tho.

**Comment by Mhowser** [comment]
@tuxun do you have a detailed way to build the client with OpenCL support so i can test it?

**Comment by tuxun** [comment]
For what I know its far to be stable. you have to remove the test mentioned in  [this answer](https://github.com/laurent-clouet/sheepit-client/issues/165#issuecomment-460585408) , then sheepit should see your card (if the proprietary amdgpu opencl stack is installed on your system). 
I dont know if I have a hardware problem, a bug in blender,  in the kernel, in Xorg/Xwayland, but if your X server shutdowns your screens while rendering,  you will probably have to reboot :/

**Comment by Mhowser** [comment]
Thanks, I will try it out later!

**Comment by tuxun** [comment]
last amd gpu driver update seems works like a charm!

**Comment by MCOfficer** [comment]
> last amd gpu driver update seems works like a charm! could someone else confirm this fact please?

can you clarify which one, the in-kernel driver for 5.9?

**Comment by tuxun** [comment]
nope, the DKMS one

**Comment by tuxun** [comment]
problem solved! Opencl under linux is now disabled server side.

**Comment by Mhowser** [comment]
So do we just update the client and it will work?

**Comment by MCOfficer** [comment]
it is disabled. No matter what client is used, the server will not assign you OpenCL jobs.

**Comment by Mhowser** [comment]
That sounds kinda useless to be honest. If I only have OpenCL capable hardware, the server will not send me OpenCL jobs? Did I misunderstand what you said?

**Comment by MCOfficer** [comment]
let me be completely clear:

If you are on linux/OSX **and** have a non-nvidia graphics card **and** modified the client to allow that card to work, you will no longer receive jobs for this card. 

These setups were never supposed to work anyways, we just changed the way it is enforced. You will of course still receive CPU jobs.

**Comment by Mhowser** [comment]
Ah thanks for clarifying.

It is still pretty unfortunate to see yet another case of vender lock-in on such a cool project...

**Comment by Raimund58** [comment]
Well, we would like to support as much hardware as possible but since this vendor doesn't produce reliable outputs, we can not allow it.

**Comment by tuxun** [comment]
That what you get when you never warn a vendor its product dont work as expected.
And now its disabled, AMD will never fix it.

**Comment by tuxun** [comment]
Also, thanks for no help and no answer lasts 4 months.

**Comment by Raimund58** [comment]
This is GitHub, not a helpdesk. 
On Discord you got your answers quite fast.

**Comment by MCOfficer** [comment]
> 
> 
> Also, thanks for no help and no answer lasts 4 months, I'm sad when i think i contributed to your stupid discord.

I'm so confused. You were always so friendly, hen we talked to you on discord and even here

> 
> 
> problem solved! Opencl under linux is now disabled server side.


... and then this? are you the only person at your keyboard?

**Comment by Mhowser** [comment]
I mean.. What else were you expecting when you dropped that bombshell on us?

**Comment by MCOfficer** [comment]
> I mean.. What else were you expecting when you dropped that bombshell on us?

Disappointment for sure. I'm a linux user myself, and my next card will be AMD, so I can sympathize. I can understand both polite disappointment (like you've shown) and straight up bitterness.

What I didn't expect was such an inconsistency in tone. They were friendly even *after* hearing the news, and left the discord on good terms (or so it seemed). 

Not to mention that this statement is rather disingenuous, since we tried to help them for months, and answers were usually given within hours.

**Comment by Seltonu** [comment]
To the project managers, I just want to say although I'm saddened by the decision to stop sending OpenCL jobs to Linux users, I understand the position. The client doesn't support it anyways so nothing really changes, but it sounds like this prevents people from even being able to work towards a fix in the future. Is this the best option?

I'm not very knowledgable on OpenCL/SheepIt and I'm a student with very limited time, but I keep up with cool projects like this that interest me a lot so that one day when I'm smarter I can contribute. I've gotten emails on this issue for months because it's something I've watched and wanted to contribute to via testing/etc. if I could be useful at all to the people working on this.

I would love the opportunity to test for people if the maintainers can find a middle ground that lets Linux users slowly work on a fix without affecting people's renders. Really I would like to put my system to use rendering for artists who need it, that's why we're all here. Especially given AMD is announcing new hardware literally today, I've been so hyped to get big Navi and really find some interesting uses for it outside of gaming. I'm still hoping sheepIt can be one of those cool uses eventually (and also keep my room warm instead of using a heater lol)

**Comment by MCOfficer** [comment]
> but it sounds like this prevents people from even being able to work towards a fix in the future. Is this the best option?

I cannot speak for Clou, but it seems possible to disable this check on the [Sandbox](https://sandbox.sheepit-renderfarm.com/), which should be used for development.

**Comment by laurent-clouet** [comment]
Yes, on the sandbox, you can use linux amd gpu, it's made for this exact purpose (development&testing)

**Comment by tuxun** [comment]
except we cant upload project on it as I reported to you everytime i tried to use it
.

**Comment by tuxun** [comment]
Oh no, storage server has crashed.

Administrators have been notified.


Oh no, storage server has crashed.

Administrators have been notified.


Oh no, storage server has crashed.

Administrators have been notified.


Oh no, storage server has crashed.

Administrators have been notified.

**Comment by tuxun** [comment]
If Administrators have been notified. why the f they dont fix the stuuf they told us to use???

**Comment by tuxun** [comment]
![image](https://user-images.githubusercontent.com/3581513/97790399-b6402b80-1bc8-11eb-8b87-fa53c014e92b.png)
![image](https://user-images.githubusercontent.com/3581513/97790464-11721e00-1bc9-11eb-801b-0f22be1c4804.png)

**Comment by tuxun** [comment]
Also sandbox dont propose blender 2.9, the blender version you keep telling me to use

**Comment by tuxun** [comment]
answer are so fast on discord that sandbox is UNUSABLE SINCE 90 days
